ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -204,8 +207,9 @@ private void resetClient(AppendEntriesRequest request, Event event) {
           .map(TermIndex::getIndex)
           .orElseGet(f::getMatchIndex);
       if (event.isError() && request == null) {
-        LOG.warn("{}: Follower failed (request=null, errorCount={}); keep nextIndex ({}) unchanged and retry.",
-            this, errorCount, f.getNextIndex());
+        BatchLogger.warn(LOG, BatchLogger.makeUniqueId(f.getId(), f.getNextIndex()), () ->
+            LOG.warn("{}: Follower failed (request=null, errorCount={}); keep nextIndex ({}) unchanged and retry.",
+                this, errorCount, f.getNextIndex()), exceptionalWarnLogBatchDuration);
```

Comment:
Store `f.getNextIndex()`; otherwise, it may change.
```java
        final long followerNextIndex = f.getNextIndex();
        BatchLogger.warn(LOG, BatchLogger.makeUniqueId(f.getId(), followerNextIndex), () ->
            LOG.warn("{}: Follower failed (request=null, errorCount={}); keep nextIndex ({}) unchanged and retry.",
                this, errorCount, followerNextIndex), exceptionalWarnLogBatchDuration);
```

Reply:
You are right, Thanks for catching this bug!